tests/processes/ProcessingTest.py

```diff
@@ -329,6 +387,25 @@ def test_get_file_dict_non_existent_file(self):
                        'an unknown error.'),
                       self.log_printer.log_queue.get().message)
 
+    def test_get_file_dict_allow_raw_file(self):
+        file_dict = get_file_dict([self.unreadable_path], self.log_printer,
+                                  True)
+        self.assertNotEqual(file_dict, {})
+        self.assertEqual(file_dict[self.unreadable_path], None)
+        self.assertNotIn(("Failed to read file '{}'. It seems to contain "
```

Comment:
An idea that I just had over at https://github.com/coala/coala/pull/4270#discussion_r121594930 , ...

It would be more useful to assert that there are no warnings (or errors) in the message queue.

And as part of that, instead of `self.log_printer.log_queue.get().message`, fetch all messages into an array, so that if another message is emitted somewhere in new code before the target message, it doesnt cause this test to silently fail.